apps/condo/domains/ticket/schema/TicketAnalyticsReportService.js

```diff
@@ -21,55 +47,60 @@ const TicketAnalyticsReportService = new GQLCustomSchema('TicketAnalyticsReportS
-                    },
-                })
-                const result = {}
-                // TODO(sitozzz): rewrite to support `n` levels of groupping
-                const [group1Name, group2Name] = groupBy
-                for (const group1Option in groupedCounters) {
-                    result[translates[group1Name][group1Option]] = {}
-                    for (const group2Option in groupedCounters[group1Option]) {
-                        result[translates[group1Name][group1Option]][translates[group2Name][group2Option]] = allTickets.filter(
-                            ticket =>
-                                get(ticket, groupByFields[group1Name]) === group1Option &&
-                                get(ticket, groupByFields[group2Name]) === group2Option
-                        ).length
+
+                const ticketGqlToKnexAdapter = new TicketGqlToKnexAdapter(where, groupBy)
+                await ticketGqlToKnexAdapter.loadData()
+
+                const translates = {}
+                for (const group of groupBy) {
+                    switch (group) {
+                        case 'property':
+                            translates[group] = await createPropertyRange(context, where.organization)
+                            break
+                        case 'status':
+                            translates[group] = await createStatusRange(context, where.organization)
+                            break
+                        default:
+                            break
                     }
                 }
+
+                const result = ticketGqlToKnexAdapter
+                    .getResult(({ count, dayGroup, ...searchResult }) =>
+                    {
+                        if (!isEmpty(translates)) {
+                            Object.entries(searchResult).forEach(([groupName, value]) => {
+                                const translateMapping = get(translates, groupName, false)
+                                if (translateMapping) {
+                                    searchResult[groupName] = translateMapping.find(translate => translate.value === value).label
+                                }
+                            })
+                            return {
+                                ...searchResult,
+                                dayGroup: moment(dayGroup).format(DATE_DISPLAY_FORMAT),
+                                count: parseInt(count),
+                            }
+                        }
+                        return {
+                            ...searchResult,
+                            dayGroup: moment(dayGroup).format(DATE_DISPLAY_FORMAT),
+                            count:parseInt(count),
+                        }
+                    }).sort((a, b) =>
+                        moment(a.dayGroup, DATE_DISPLAY_FORMAT).format('X') - moment(b.dayGroup, DATE_DISPLAY_FORMAT).format('X'))
                 return { result }
```

Comment:
You will have `data.result.result` in output, let it be `data.result.groups`, for example.

```suggestion
                return { groups: result }
```

Reply:
Done